snippet_ranger/models/snippet.py

```diff
@@ -17,19 +17,32 @@ def construct(self, repository, filenames, sources, uasts,
         super(Snippet, self).construct(repository=repository, filenames=filenames,
                                        sources=sources, uasts=uasts)
         if ((positions_start is None) ^ (positions_end is None)) or \
```

Comment:
it can be rewritten as:
```
if (position_start is None) and ((positions_end is None) or (positions is not None)):
```
because `(position_start is None)` is contained in both conditions.
Just to make it a bit shorter

Reply:
it is better to leave as is because one can see the logic of the expression. And I think it is not equal, because ^ is XOR operation.

Reply:
yes, you are right